src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -351,6 +353,21 @@ trait ClusterComponent extends LeoComponent {
       }
     }
 
+    def clearKernelFoundBusyDate(id: Long): DBIO[Int] = {
+      findByIdQuery(id).map(_.kernelFoundBusyDate).update(None)
+    }
+
+    def updateKernelFoundBusyDate(id: Long, kernelFoundBusyDate: Instant): DBIO[Int] = {
+      findByIdQuery(id).map(_.kernelFoundBusyDate).update(Option(Timestamp.from(kernelFoundBusyDate)))
+    }
+
+    def clearKernelFoundBusyDateByProjectAndName(googleProject: GoogleProject, clusterName: ClusterName): DBIO[Int] = {
+      clusterQuery.getActiveClusterByName(googleProject, clusterName) flatMap {
```

Comment:
Maybe this could call `getActiveClusterByNameMinimal` instead to avoid all the left joins.